unitTests/ui/StarRatingTest.ts

```diff
@@ -0,0 +1,155 @@
+import { l } from '../../src/Core';
+import { FakeResults } from '../Fake';
+import { $$ } from '../../src/utils/Dom';
+import * as Mock from '../MockEnvironment';
+import { IQueryResult } from '../../src/rest/QueryResult';
+import { StarRating, IStarRatingOptions } from '../../src/ui/StarRating/StarRating';
+
+const CONTAINER_CSS_CLASS = 'CoveoStarRating';
+const STAR_CSS_CLASS = 'coveo-star-rating-star';
+const ACTIVE_STAR_CSS_CLASS = 'coveo-star-rating-star-active';
+const STAR_LABEL_CSS_CLASS = 'coveo-star-rating-label';
+
+const getActiveStars = (starRatingElement: HTMLElement) => {
+  const children = starRatingElement.children;
+  let numStars = 0;
+  let numActiveStars = 0;
+  let i = 0;
+  for (; i < children.length; i++) {
+    if (children.item(i).classList.contains(STAR_CSS_CLASS)) {
+      if (children.item(i).classList.contains(ACTIVE_STAR_CSS_CLASS)) {
+        numActiveStars++;
+      }
+      numStars++;
+    }
+  }
+  return { numStars: numStars, numActiveStars: numActiveStars };
+};
+
+export function StarRatingTest() {
+  describe('StarRating', () => {
+    let test: Mock.IBasicComponentSetup<StarRating>;
+
+    function initStarRatingComponent(ratingValue: String, numRatingsValue?: String, ratingScale?: number) {
+      const options: IStarRatingOptions = { rating: '@rating', numberOfRatings: '@numberOfRatings', ratingScale: ratingScale };
```

Comment:
Could be simplified as : 
```
const options = { rating: '@rating', numberOfRatings: '@numberOfRatings', ratingScale };
const result = FakeResults.createFakeResult();
result.raw.rating = ratingValue;
result.raw.numberOfRatings = numRatingsValue;

test = Mock.optionsResultComponentSetup<StarRating>(StarRating, result, options)
```

Reply:
I would also prefer to not pass options as args to the function. Rather, create an `options` variable on line 32 that is reset to default values in a beforeEach. Then in your test, adjust the options before calling `initStarRatingComponent`.